modules/photo/src/chromatic_aberration_correction.cpp

```diff
@@ -0,0 +1,190 @@
+#include "precomp.hpp"
+
+#include <iostream>
+
+namespace cv {
+
+bool CalibrationResult::loadFromFile(const String& filename) {
+    try {
+        FileStorage fs(filename, FileStorage::READ);
+        if (!fs.isOpened()) {
+            std::cerr << "Trying to open: " << filename << std::endl;
+            return false;
+        }
+        
+        fs["degree"] >> degree;
+        FileNode red_node = fs["red_channel"];
+        poly_red.degree = degree;
+        red_node["coeffs_x"] >> poly_red.coeffs_x;
+        red_node["coeffs_y"] >> poly_red.coeffs_y;
+        red_node["mean_x"] >> poly_red.mean_x;
+        red_node["mean_y"] >> poly_red.mean_y;
+        red_node["std_x"] >> poly_red.std_x;
+        red_node["std_y"] >> poly_red.std_y;
+        
+        FileNode blue_node = fs["blue_channel"];
+        poly_blue.degree = degree;
+        blue_node["coeffs_x"] >> poly_blue.coeffs_x;
+        blue_node["coeffs_y"] >> poly_blue.coeffs_y;
+        blue_node["mean_x"] >> poly_blue.mean_x;
+        blue_node["mean_y"] >> poly_blue.mean_y;
+        blue_node["std_x"] >> poly_blue.std_x;
+        blue_node["std_y"] >> poly_blue.std_y;
+        
+        fs.release();
+        return true;
+    } catch (const Exception& e) {
+        std::cerr << "Trying to open: " << filename << std::endl;
+        std::cout << "Error loading calibration file: " << e.what() << "\n";
+        return false;
+    }
+}
+
+void Polynomial2D::computeDeltas(const Mat& X, const Mat& Y, Mat& dx, Mat& dy) const {
+    CV_Assert(X.type() == CV_32F && Y.type() == CV_32F && X.size() == Y.size());
+
+    const int h = X.rows, w = X.cols, D = degree;
+    dx.create(X.size(), CV_32F);
+    dy.create(Y.size(), CV_32F);
+
+    const double inv_std_x = 1.0 / std_x;
+    const double inv_std_y = 1.0 / std_y;
+
+    parallel_for_( Range(0, h),
+        [&](const Range& rows)
+    {
+        std::vector<double> x_pow(D + 1);
+        std::vector<double> y_pow(D + 1);
+
+        for (int y = rows.start; y < rows.end; ++y)
+        {
+            const float* XR = X.ptr<float>(y);
+            const float* YR = Y.ptr<float>(y);
+            float* DX = dx.ptr<float>(y);
+            float* DY = dy.ptr<float>(y);
+
+            for (int x = 0; x < w; ++x)
+            {
+                const double xn = (XR[x] - mean_x) * inv_std_x;
+                const double yn = (YR[x] - mean_y) * inv_std_y;
+
+                x_pow[0] = y_pow[0] = 1.0;
+                for (int k = 1; k <= D; ++k)
```

Comment:
I think, the 2 main sources of speedup would be:

1) migrate from double precision to single precision (if possible).
2) fix the degree so that we can unroll the loop and just use Horner's method instead of those arrays with powers.
3) use OpenCV universal intrinsics and unroll the loop by x, so that we can evaluate polynomials for several pixels simultaneously (e.g. do it for 8 pixels in the case of Intel AVX2 and for 4 pixels in the case of ARM Neon).
4) embed bilinear interpolation right into the same loop that computes polynomials, don't store huge maps.